scala-interpreter/src/main/scala-2.10/org/apache/toree/kernel/interpreter/scala/ScalaInterpreterSpecific.scala

```diff
@@ -61,6 +61,7 @@ trait ScalaInterpreterSpecific { this: ScalaInterpreter =>
   ): SparkIMain = {
     val s = new SparkIMain(settings, out)
     s.initializeSynchronous()
+    System.setProperty("spark.repl.class.outputDir", s.getClassOutputDirectory.getAbsolutePath)
```

Comment:
I've tried to figure out how to do that without heavy code modifications but haven't succeeded.

It worth noting that currently in a Scala 2.11 environment Toree [sets this property in a similar manner](https://github.com/apache/incubator-toree/blob/master/kernel/src/main/scala/org/apache/toree/boot/KernelBootstrap.scala#L70). And in Scala 2.11 we can pass it directly to `SparkConf` with ease because `spark.repl.class.outputDir`'s required value becomes known very early ([when `org.apache.spark.repl.Main` is initialised](https://github.com/apache/spark/blob/branch-1.6/repl/scala-2.11/src/main/scala/org/apache/spark/repl/Main.scala#L33)).

But in a Scala 2.10 environment, things are different. `spark.repl.class.outputDir`' [is initialised only with `org.apache.spark.repl.SparkIMain`](https://github.com/apache/spark/blob/branch-2.2/repl/scala-2.10/src/main/scala/org/apache/spark/repl/SparkIMain.scala#L100), that is after creation of `ScalaInterpreter` in Toree. But in order to pass `spark.repl.class.outputDir` directly to `SparkConf` we have to initialise`SparkIMain` earlier than Toree's `Kernel`.

This is applicable for both Spark 2 and Spark 1.6.